stdlib/REPL/test/repl.jl

```diff
@@ -51,7 +51,10 @@ function fake_repl(@nospecialize(f); options::REPL.Options=REPL.Options(confirm_
     repl.options = options
 
     hard_kill = kill_timer(900) # Your debugging session starts now. You have 15 minutes. Go.
+    # @eval Base old_have_color = have_color
+    # @eval Base have_color = false
```

Comment:
To pass this test we have to disable colors. If not this way, then we need another mechanism.

Reply:
@karan0299 I think you will have to implement a `set(::TTY, key::Symbol, value::Bool)` then we can disable color for `fake_repl`

Reply:
@karan0299 This was tricky and required adding `hascolor` to the concrete REPL types. That done all references to `Base.have_color` are removed from the REPL package: 59586de .

Feel free to merge from https://github.com/mkitti/julia/tree/TTY-STDOUT-color

Reply:
I’m so grateful for your help. Thank you very much.

Reply:
Starting to look good! So much nicer without the global. Can this now be done by passing `false` to the REPL constructors above?

Reply:
@JeffBezanson 

For `LineEditREPL` you can now just pass in `hascolor` as I do for the test above. The constructor previously ignored it and always set it to `true`.

`BasicREPL` does not have a struct field to store `hascolor`. It purely depends on `hascolor(r.terminal)`.
Similarly, `StreamREPL` now purely depends on `hascolor(r.stream)`.

I did not implement `hascolor(r::AbstractREPL)`. Should we? It would probably be `hascolor(r::AbstractREPL) = false`

I think there is a PR for another day where one eliminates the redundancy of `hascolor` between `REPL.Options` and the `LineEditREPL`. We could also consider how `REPL.GlobalOptions` interacts with the `BasicREPL` and `StreamREPL`.

Reply:
The tests are passing so it seems fine now. Can we remove the commented-out lines?

Reply:
@karan0299 Removing those comments is all you. I already removed them in my branch.